### ISSUE TITLE
[TW#28515] http_server:  "server_port" overide settings Not working

### ISSUE THREAD

**Issue by versamodule** [espressif/esp-idf#2988]
when I change the server_port from the default vaule of 80, it does not work on that port, but yet works on port 80 no matter what I set it to. 

httpd_handle_t web_server = NULL;
httpd_config_t config = HTTPD_DEFAULT_CONFIG();
config.server_port = 8080;
if (httpd_start(&web_server, &config) == ESP_OK) {...}

**Comment by mahavirj** [comment]
@versamodule Could you please send across device log and also IDF commit/version (as per issue template)?

**Comment by versamodule** [comment]
I wanted to but I am using VisualGDB and I do not have access to the terminal to issue commands to find this information out.  They say they are using 3.2. 

![8b100b02-5829-4134-bd0a-b0fe2435a8c5 png](https://user-images.githubusercontent.com/6456652/51747587-8c5eea00-206f-11e9-9732-28f4b980bc8d.jpg)

**Comment by mahavirj** [comment]
@versamodule Unfortunately I am unable to reproduce this, if we walk through code, `server_port` is definitely being used,https://github.com/espressif/esp-idf/blob/18118a6d5ceebef59127ab3d2319f616c2924880/components/esp_http_server/src/httpd_main.c#L242

Can you check around this through debug logs at your end?

**Comment by mahavirj** [comment]
@versamodule Did you get chance to look into this?

**Comment by versamodule** [comment]
Hello, I'll create a short demo to show it works when set to port 80, but when I change the port it stops.

**Comment by versamodule** [comment]
I am speechless. I took the time to prove this was not working before I posted the problem to avoid anyone wasting their time. I changed "config.server_port " from 80 to 8080 and back several times. Even doing a clean build. It was always repeatable. 
I created a new project today, and same code worked find on both ports. I just don't know what to say other than thank you for your time, and I am sorry I wasted it.
